contracts/programs/ocr2/src/lib.rs

```diff
@@ -783,30 +789,44 @@ impl Report {
     }
 }
 
-fn calculate_reimbursement(juels_per_lamport: u64, _signature_count: usize) -> Result<u64> {
-    const SIGNERS: u64 = 1;
+fn calculate_reimbursement_gjuels(juels_per_lamport: u64, _signature_count: usize) -> Result<u64> {
+    const SIGNERS: u128 = 1;
+    const GIGA: u128 = 10u128.pow(9);
     let fees = Fees::get()?;
-    let lamports_per_signature = fees.fee_calculator.lamports_per_signature;
+    let lamports_per_signature: u128 = fees.fee_calculator.lamports_per_signature.into();
     let lamports = lamports_per_signature * SIGNERS;
-    let juels = lamports * juels_per_lamport;
-    Ok(juels)
+    let juels = lamports * u128::from(juels_per_lamport);
+    let gjuels = juels / GIGA; // return value as gjuels
+
+    // convert from u128 to u64 with staturating logic to max u64
+    let convert = gjuels.try_into();
+    let gjuels_u64: u64 = match convert {
+        Ok(convert) => convert,
+        Err(_error) => u64::MAX,
+    };
+
+    Ok(gjuels_u64)
```

Comment:
Can be simplified into:

```suggestion
    Ok(gjuels.try_into().unwrap_or(u64::MAX))
```